packages/atomic/src/components/common/interface/result.ts

```diff
@@ -0,0 +1,7 @@
+import {FoldedResult, Result} from '@coveo/headless';
+
+export type AnyResult = Result | FoldedResult;
+
+export function extractFoldedResult(anyResult: AnyResult) {
+  return (anyResult as FoldedResult).result || anyResult;
+}
```

Comment:
Since we're actually trying to get the result *out* of a folded result, I think this should be named `extractUnfoldedResult`.